NDTensors/src/lib/GradedAxes/src/fusion.jl

```diff
@@ -53,21 +68,28 @@ function fuse_blocklengths(x::LabelledInteger, y::LabelledInteger)
   return labelled(unlabel(x) * unlabel(y), fuse_labels(label(x), label(y)))
 end
 
+flatten_maybe_nested(v::Vector{<:Integer}) = v
+flatten_maybe_nested(v::Vector{<:AbstractGradedUnitRange}) = reduce(vcat, blocklengths.(v))
```

Comment:
Could you explain this a bit? I'm confused why you might have either a vector of integers or graded unit ranges as inputs (that's strange from a generic coding perspective, those inputs are pretty different from each other), and also I'm a bit confused about what this outputs. Also I would think a better name for this might be `flatten_blocklengths`.

Reply:
I see, that sounds like a strange design choice to have that function output such different formats for different inputs. I think it would be good to redesign that, I don't remember that discussion and can't imagine why it would have been designed that way.

Reply:
@emstoudenmire would be good to get your input on this part of the code.

Reply:
This is fixed so I'm resolving this comment thread.